linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_password_pam_unix_remember/ansible/shared.yml

```diff
@@ -5,26 +5,175 @@
+  register: result_authselect_custom_profile_present
+  changed_when: false
+  when:
+    - result_authselect_present.stat.exists
+    - authselect_current_profile is not match("custom/")
+
+- name: Create a custom profile based on the current profile
+  ansible.builtin.command:
+    cmd: authselect create-profile hardening -b sssd
+  register: result_authselect_create_profile
+  when:
+    - result_authselect_present.stat.exists
+    - result_authselect_check_cmd is success
+    - authselect_current_profile is not match("custom/")
+    - not result_authselect_custom_profile_present.stat.exists
+
+- name: Ensure the desired configuration is updated in the custom profile
+  ansible.builtin.replace:
+    dest: "{{ item }}"
+    regexp: '(.*pam_pwhistory.so.*remember=)(\S+)(.*)$'
+    replace: '\g<1>{{ var_password_pam_unix_remember }}\g<3>'
+  loop:
+    - /etc/authselect/{{ authselect_custom_profile }}/system-auth
+    - /etc/authselect/{{ authselect_custom_profile }}/password-auth
+  when:
+    - result_authselect_profile is not skipped
+    - authselect_custom_profile is match("custom/")
+
+- name: Ensure the desired configuration in present in the custom profile
+  ansible.builtin.lineinfile:
+    dest: "{{ item }}"
+    insertafter: ^password.*requisite.*pam_pwquality.so.*
+    line: "password    requisite     pam_pwhistory.so remember={{ var_password_pam_unix_remember }} use_authtok"
+  loop:
+    - /etc/authselect/{{ authselect_custom_profile }}/system-auth
+    - /etc/authselect/{{ authselect_custom_profile }}/password-auth
+  when:
+    - result_authselect_profile is not skipped
+    - authselect_custom_profile is match("custom/")
+
+- name: Ensure a backup of current authselect profile before select the custom profile
+  ansible.builtin.command:
+    cmd: authselect apply-changes -b --backup=before-pwhistory-hardening.backup
+  register: result_authselect_backup
+  when:
+    - result_authselect_check_cmd is success
+    - result_authselect_profile is not skipped
+    - authselect_current_profile is not match("custom/")
+    - authselect_custom_profile is not match(authselect_current_profile)
+
+- name: Ensure the custom profile is selected
+  ansible.builtin.command:
+    cmd: authselect select {{ authselect_custom_profile }} --force
+  register: result_pam_authselect_select_profile
+  when:
+    - result_authselect_check_cmd is success
+    - result_authselect_profile is not skipped
+    - authselect_current_profile is not match("custom/")
+    - authselect_custom_profile is not match(authselect_current_profile)
+
+- name: Restore the authselect features in the custom profile
+  ansible.builtin.command:
+    cmd: authselect enable-feature {{ item }}
+  register: result_pam_authselect_select_features
+  loop: "{{ result_authselect_features.stdout_lines }}"
+  when:
+    - result_authselect_profile is not skipped
+    - result_authselect_features is not skipped
+    - result_pam_authselect_select_profile is not skipped
```

Comment:
@marcusburghardt  This remediation is reverting manual changes done to PAM files